packages/test-sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/UserTests.kt

```diff
@@ -111,6 +111,34 @@ class UserTests {
         }
     }
 
+    @Test
+    fun getProviderType() = runBlocking {
+        val email = randomEmail()
+        val emailUser = createUserAndLogin(email, "123456")
+        assertEquals(AuthenticationProvider.EMAIL_PASSWORD, emailUser.provider)
+    }
+
+    @Test
+    fun getAccessToken() = runBlocking {
```

Comment:
I would also test what this returns when the user is logged out...I'm actually not sure. I suspect it still returns the token, which the server will just reject. But capturing the behaviour would be nice